Content.Server/Automod/Actions/TextAutomodActionBan.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Content.Server.Administration.Managers;
+using Content.Shared.Automod;
+using Content.Shared.Database;
+using Robust.Shared.Player;
+
+namespace Content.Server.Automod.Actions;
+
+public sealed partial class TextAutomodActionBan : ITextAutomodAction
+{
+    [DataField]
+    public uint? BanTime;
+
+    [DataField]
+    public string Reason = "automod-action-ban-reason";
+
+    [DataField]
+    public string BanCountGroup = "default";
+
+    public bool Skip(string fullText, Dictionary<string, int> patternMatches)
+    {
+        return false;
+    }
+
+    public bool RunAction(ICommonSession session,
+        string fullText,
+        Dictionary<string, int> patternMatches,
+        AutomodFilterDef filter,
+        string filterDisplayName,
+        IEntityManager entMan)
+    {
+        var banManager = IoCManager.Resolve<IBanManager>();
+
+        // TODO ShadowCommander implement ban group tracking for banning after three filter hits within $time minutes
+
+        var str = Loc.GetString(Reason, ("filterName", filterDisplayName));
+
+        banManager.CreateServerBan(session.UserId, null, null, null, null, BanTime, NoteSeverity.Minor, str);
```

Comment:
You are not awaiting this task or handling any errors.

Reply:
I'm having trouble figuring out how to do this. How do I await it and what errors do I need to handle?

Reply:
Easiest thing is just to do something similar to event bus where managers subscribe to the relevant event based on the type in question. All these events can just be put in a partial of automod action manager to keep stuff simple.

Reply:
Call it from an `async void` and try catch it as usual.

Reply:
BanManager.CreateServerBan is an `async void` method. None of the other usages of it await calling CreateServerBan. Do I still need to do that here?